openpype/hosts/maya/plugins/publish/extract_import_reference.py

```diff
@@ -0,0 +1,156 @@
+import os
+import sys
+
+from maya import cmds
+
+import pyblish.api
+import tempfile
+
+from openpype.lib import run_subprocess
+from openpype.pipeline import publish
+from openpype.hosts.maya.api import lib
+
+
+class ExtractImportReference(publish.Extractor):
+    """
+
+        Extract the scene with imported reference.
+        The temp scene with imported reference is
+        published for rendering if this extractor is activated
+
+    """
+
+    label = "Extract Import Reference"
+    order = pyblish.api.ExtractorOrder - 0.48
+    hosts = ["maya"]
+    families = ["renderlayer", "workfile"]
+    optional = True
+    tmp_format = "_tmp"
+
+    @classmethod
+    def apply_settings(cls, project_setting, system_settings):
+        cls.active = project_setting["deadline"]["publish"]["MayaSubmitDeadline"]["import_reference"] # noqa
+
+    def process(self, instance):
+        ext_mapping = (
+            instance.context.data["project_settings"]["maya"]["ext_mapping"]
+        )
+        if ext_mapping:
+            self.log.info("Looking in settings for scene type ...")
+            # use extension mapping for first family found
+            for family in self.families:
+                try:
+                    self.scene_type = ext_mapping[family]
+                    self.log.info(
+                        "Using {} as scene type".format(self.scene_type))
+                    break
+
+                except KeyError:
+                    # set scene type to ma
+                    self.scene_type = "ma"
+
+        _scene_type = ("mayaAscii"
+                       if self.scene_type == "ma"
+                       else "mayaBinary")
+
+        dir_path = self.staging_dir(instance)
+        # named the file with imported reference
+        if instance.name == "Main":
+            return
+        tmp_name = instance.name + self.tmp_format
+        current_name = cmds.file(query=True, sceneName=True)
+        ref_scene_name = "{0}.{1}".format(tmp_name, self.scene_type)
+
+        reference_path = os.path.join(dir_path, ref_scene_name)
+        tmp_path = os.path.dirname(current_name) + "/" + ref_scene_name
+
+        self.log.info("Performing extraction..")
+
+        # This generates script for mayapy to take care of reference
+        # importing outside current session. It is passing current scene
+        # name and destination scene name.
+        script = ("""
+# -*- coding: utf-8 -*-
+'''Script to import references to given scene.'''
+import maya.standalone
+maya.standalone.initialize()
+# scene names filled by caller
+current_name = "{current_name}"
+ref_scene_name = "{ref_scene_name}"
+print(">>> Opening {{}} ...".format(current_name))
+cmds.file(current_name, open=True, force=True)
+print(">>> Processing references")
+all_reference = cmds.file(q=True, reference=True) or []
+for ref in all_reference:
+    if cmds.referenceQuery(ref, f=True, il=True):
```

Comment:
It seems maya is bitch:

```
RuntimeError: These flags are mutually exclusive (except for 'ea/editAttrs' and 'en/editNodes' which may be used together): 'f/filename', 'rfn/referenceNode', 'inr/isNodeReferenced', 'n/nodes', 'es/editStrings', 'en/editNodes', and 'ea/editAttrs'. 
```

We need to drop the `f` flag here.

```suggestion
    if cmds.referenceQuery(ref, il=True):
```